### PR TITLE
Ruby group capstone - Catalog of my things

### PR DESCRIPTION
**Pull request by tomdan-ai**
# In This Ruby Capstone Project
#### Logic

- Each class from the [UML class diagram](./images/catalog_of_my_things.png) are created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes are all implemented.
- All properties visible in the diagram are defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships are NOT set in the constructor method. Instead, they have a custom setter method created._
- All methods visible in the diagram are well implemented:
    - add_item method (in all classes that have that method)
        - takes an instance of the Item class as an input.
        - adds the input item to the collection of items.
        - adds self as a property of the item object (by using the correct setter from the item object).
    - can_be_archived?() in the Item class
        - returns true if published_date is older than 10 years.
        - otherwise, it returns false.
    - move_to_archive() in the Item class
        - reuses can_be_archived?() method.
        - changes the archived property to true if the result of the can_be_archived?() method is true.
        - does nothing if the result of the can_be_archived?() method is false.
    - can_be_archived?() in the Book class
        - overrides the method from the parent class.
        - returns true if the parent's method returns true OR if cover_state equals to "bad".
        - otherwise, it returns false.
    - can_be_archived?() in the MusicAlbum class
        - overrides the method from the parent class.
        - returns true if parent's method returns true AND if on_spotify equals true.
        - otherwise, it returns false.
    - can_be_archived?() in the Movie class
        - overrides the method from the parent class.
        - returns true if parent's method returns true OR if silent equals true.
        - otherwise, it returns false.
    - can_be_archived?() in the Game class
        - overrides the method from the parent class.
        - returns true if parent's method returns true AND if last_played_at is older than 2 years.
        - otherwise, it returns false.
- Add unit tests for all implemented methods.

#### User interface

- Create a main.rb file that serves as the console app entry-point.
- Your console app, at the start:
    - Present the user with a list of options to perform.
    - Let users choose an option.
    - ask for parameters for the option.
    - Have a way to quit the app.
- The following options are available:
    - List all books
    - List all music albums
    - List all movies
    - List of games
    - List all genres (e.g 'Comedy', 'Thriller')
    - List all labels (e.g. 'Gift', 'New')
    - List all authors (e.g. 'Stephen King')
    - List all sources (e.g. 'From a friend, 'Online shop')
    - Add a book
    - Add a music album
    - Add a movie
    - Add a game
 - All data is preserved by saving collections in .json files.

#### Database
- Create a schema.sql file with tables to be analogical to the structure of the classes in the app. As you cannot implement inheritance in the database tables 
- Add all properties and associations from the parent Item class as table columns to all tables based on the child classes.

### CODE REVIEW

File: genre.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Genre
+  attr_accessor :name, :items
+
+  def initialize(name)
+    @name = name
+    @items = []
+  end
```
Comment (AlphaNtihinduka) [inline genre.rb]: - [ ] According to the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png), the genre should have an id. This is to ensure that each genre is uniquely created ☺️.

File: data/labels.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+[
+  {
+    "title": "vrger",
+    "color": "vsrgew"
+  },
```
Comment (AlphaNtihinduka) [inline data/labels.json:5]: - [ ] According to the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png), the label should have an id. This is to ensure that each label is uniquely created ☺️.

**Review by AlphaNtihinduka — CHANGES_REQUESTED** [review]
## Hi @tomdan-ai 🙋‍♂️
Good job working on . You have done incredibly well. 👏  

You still need to implement few on the project but you are almost there!!! 🚶: 

TO HIGHLIGHT:
- Good UI ✔️ 
- Proper use of inheritance ✔️ 
- Well-structured files ✅

# STATUS: Required Changes :recycle: 
- Kindly consider the inline review comments ☺️
-  [ ] In addition, kindly ensure that the video does not go beyond 5 minutes as described in the project requirements. Also, @Bellagirl-maker should also include in the code is proud of ☺️.

close

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi Team,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**